service/src/test/java/uk/gov/hmcts/reform/fpl/service/ManageHearingsServiceTest.java

```diff
@@ -1145,9 +1145,45 @@ void shouldAddNewHearing() {
             service.addOrUpdate(newHearing, caseData);
 
             assertThat(caseData.getHearingDetails())
-                .containsExactly(hearing1, hearing2, newHearing);
+                .containsExactly(newHearing, hearing2, hearing1);
+        }
+
+        @Test
+        void shouldSortByHearingStartDateWhenSameHearingDayExist() {
+            LocalDateTime timeNow = time.now();
+            Element<HearingBooking> hearing1 = element(hearing(timeNow.plusDays(1), time.now().plusDays(2)));
+            Element<HearingBooking> hearing2 = element(hearing(timeNow.plusDays(2), time.now().plusDays(3)));
+            Element<HearingBooking> hearing3 = element(hearing(timeNow.plusDays(3), time.now().plusDays(4)));
+            Element<HearingBooking> newHearing = element(hearing(timeNow.plusDays(3), time.now().plusDays(4)));
+
+            CaseData caseData = CaseData.builder()
+                .hearingDetails(newArrayList(hearing2, hearing3, hearing1))
+                .build();
+
+            service.addOrUpdate(newHearing, caseData);
+
+            assertThat(caseData.getHearingDetails())
+                .containsExactly(hearing3, newHearing, hearing2, hearing1);
         }
 
+        @Test
+        void shouldSortByHearingStartDateWhenExistingHearingAreAlreadySorted() {
+            LocalDateTime timeNow = time.now();
+            Element<HearingBooking> hearing1 = element(hearing(timeNow.plusDays(1), time.now().plusDays(2)));
+            Element<HearingBooking> hearing2 = element(hearing(timeNow.plusHours(1), time.now().plusDays(4)));
+            Element<HearingBooking> hearing3 = element(hearing(timeNow.plusDays(2), time.now().plusDays(3)));
+            Element<HearingBooking> hearing4 = element(hearing(timeNow.plusDays(4), time.now().plusDays(5)));
+            Element<HearingBooking> newHearing = element(hearing(timeNow.plusDays(4), time.now().plusDays(5)));
```

Comment:
Looks new `hearing4` and `newHearing` having the same hearing date? I am wondering how it will handle it if the hearing date is the same.

Reply:
if same dates, those hearing will stick to existing ordering behaviour., ie. ordered by hearing number.